### ISSUE TITLE
Connection to heartbeat server should match protocol of web app

### ISSUE THREAD

**Issue by elondaits** [hilbert/hilbert-heartbeat#7]
When you connect to a web server through https (e.g. a web app running on a station, accessed through https) all other connections made in that page have to be https as well, or they get blocked by the browser. I *think* this would include the XMLHttpRequest connection made by the heartbeat js, since even images have to be accessed through http.

Consider also:
- There's a big push towards using https everywhere.
- Google Chrome 56 will mark http connections as insecure if the page has password or credit card fields and eventually move on to mark all http connections as insecure.
- Museum IT could choose to serve its sites through https for their own reasons, even using a free self signed cert.

So the possibility of the heartbeat server running over https has to be taken into consideration, as it might be required.

My recommendation:
- Change example URLs and default URLs from 'http://example.com' to '//example.com'. Browsers interpret URLs starting with '//' as 'connect through the same protocol used to get this page' and will use http or https as appropriate.

**Comment by porst17** [comment]
I don't think this makes much sense in the setup `heartbeat.js` is supposed to be used. `heartbeat.js` may be loaded from an exhibit-local website (e.g. via `http://localhost[:PORT]/...`), via `file://` if is is loaded from within Electron or whatever is appropriate (I can think of usage within`nodejs`...). 

However, the heartbeat server is always talking `http[s]://`, regardless of the protocoll used to load the website `heartbeart.js` is in.

At least for `heartbeat.js`, it is possible to supply any URL that is working with `XMLHttpRequest` to connect to the heartbeat server, i.e. providing `//[HOST]:[PORT][/PATH]` should be totally OK (although it is untested).

I'll leave `heartbeat.js` unchanged. But it looks like the Bash client and Python server are currently stuck at `http://` because it is hardcoded.

@malex984 Can you add support for `https://` to the Bash client? Since you are using `curl` internally, you could just switch from `hb.sh HOST PORT` to `hb.sh URL`.

**Comment by elondaits** [comment]
So, to simplify this issue: If your app is running https then the heartbeat should use https as well, otherwise you'll get an error. So basically it should always use https or detect the current protocol and select the matching one.

**Comment by porst17** [comment]
Note for myself: In JS, parsing the URL can be done by the integrated parser of the browser:
```JavaScript
//  create an anchor element (note: no need to append this element to the document)
var link = document.createElement('a');
//  set href to any path
link.setAttribute('href', 'http://example.com:12345/blog/foo/bar?startIndex=1&pageSize=10');

// The browser's built-in parser has already done its job. Now you can just grab the parts you need:

//  get any piece of the url you're interested in
link.hostname;  //  'example.com'
link.port;      //  12345
link.search;    //  '?startIndex=1&pageSize=10'
link.pathname;  //  '/blog/foo/bar'
link.protocol;  //  'http:'

//  cleanup for garbage collection
link = null;
```
This should also work for URLs starting with `//`.

**Comment by porst17** [comment]
@elondaits Please take a quick look at #15 if this is what we were talking about. @malex984 Can you test it somehow?

**Comment by elondaits** [comment]
Seems OK... would just need to be tested.

**Comment by porst17** [comment]
Fixed via hilbert/hilbert-heartbeat#15
